pkg/autodiscover/autodiscover_operators.go

```diff
@@ -63,32 +64,61 @@ func isIstioServiceMeshInstalled(allNs []string) bool {
 	return true
 }
 
-func findOperatorsByLabel(olmClient clientOlm.Interface, labels []labelObject, namespaces []configuration.Namespace) []*olmv1Alpha.ClusterServiceVersion {
-	csvs := []*olmv1Alpha.ClusterServiceVersion{}
-	for _, ns := range namespaces {
-		for _, aLabelObject := range labels {
-			label := aLabelObject.LabelKey + "=" + aLabelObject.LabelValue
-			log.Debug("Searching CSVs in namespace %q with label %q", ns, label)
-			csvList, err := olmClient.OperatorsV1alpha1().ClusterServiceVersions(ns.Name).List(context.TODO(), metav1.ListOptions{
-				LabelSelector: label,
-			})
-			if err != nil {
-				log.Error("Error when listing csvs in namespace %q with label %q", ns, label)
-				continue
-			}
-
-			for i := range csvList.Items {
-				csvs = append(csvs, &csvList.Items[i])
-			}
+func findLabelSelector(labels []labelObject) map[string]string {
```

Comment:
why convert []labelObject to map[string]string? this is not needed

Reply:
To make it easy for me to retrieve data, but I agree with you that I can use LabelObject.